File: Server/WebAppWithServer copy/server/BeatStream.js
exports.BeatStream = function () {
    this.lastIndexSent = 0;
    this.setToDefault = function () {
        this.timeForNextBeat = new Array(1000);
        this.amplitudeForNextBeat = new Array(1000);
        this.drumNumber = new Array(1000);
        this.lastIndexSent = 0;
        for (var counter = 0; counter < this.timeForNextBeat.length; counter++) {
            this.timeForNextBeat[counter] = 5000;
            this.amplitudeForNextBeat[counter] = 4;
            this.drumNumber[counter] = 0;
        }
    };

    this.createNewBeatStream = function (data) {
        var beatFrameLength = data.length;
        this.timeForNextBeat = new Array(beatFrameLength);
        this.amplitudeForNextBeat = new Array(beatFrameLength);
        this.drumNumber = new Array(beatFrameLength);
        for (var counter = 0; counter < this.timeForNextBeat.length; counter++) {
            this.timeForNextBeat[counter] = data[counter][0];
            this.amplitudeForNextBeat[counter] = data[counter][1];
            this.drumNumber[counter] = data[counter][2];
        }
    };

    this.getBeat = function (index) {
        var result = new Array(4);
        var beatIndex = index;  
        result[0] = index;
        result[1] = this.timeForNextBeat[index];
        result[2] = this.amplitudeForNextBeat[index];
        result[3] = this.drumNumber[index];
        console.info("Result[0], Result[1] : " + result[0] + ", " + result[1]);
        this.lastIndexSent = beatIndex;
        return result;
    };

    this.getBeatArray = function (index) {
        var arrayLength = this.timeForNextBeat.length - index;
        var result = new Array(arrayLength);
        for (var beatCounter = 0; beatCounter < arrayLength; beatCounter++) {
            result[beatCounter] = this.getBeat(beatCounter + index);
        }
        return result;
    };

    this.setToDefault();

};

File: Android/DrummerBot/app/src/main/java/com/thoughtworks/drummerbot/drummerbot/BeatStream.java
package com.thoughtworks.drummerbot.drummerbot;

import java.io.BufferedOutputStream;
import java.nio.ByteBuffer;
import java.sql.SQLOutput;

public class BeatStream {
    public short[] timeForNextBeat;//limitation on higher limit
    public short[] amplitudeForNextBeat;
    public short[] drumNumber;
    public int lastIndexSent = 0;
    public int beatFrameLength;
    private int[][] beats;

    public BeatStream() {
        setToDefault();
    }

    public void setToDefault() {
        this.timeForNextBeat = new short[1000];
        this.amplitudeForNextBeat = new short[1000];
        this.drumNumber = new short[1000];
        this.lastIndexSent = 0;
        for (int ii = 0; ii < timeForNextBeat.length; ii++) {
            timeForNextBeat[ii] = 5000;
            amplitudeForNextBeat[ii] = 4;
            drumNumber[ii] = 0;
        }
    }

    public byte[] getBeat(short index) {
        byte[] result = new byte[8];
        short beatIndex = index;
        result[0] = (byte) (index % 128);
        result[1] = (byte) (index / 128);

        System.out.println("Result[0], Result[1] : " + result[0] + ", " + result[1]);

        result[2] = (byte) (timeForNextBeat[beatIndex] % 128);
        result[3] = (byte) (timeForNextBeat[beatIndex] / 128);
        result[4] = (byte) (amplitudeForNextBeat[beatIndex] % 128);
        result[5] = (byte) (amplitudeForNextBeat[beatIndex] / 128);
        result[6] = (byte) (drumNumber[beatIndex] % 128);
        result[7] = (byte) (drumNumber[beatIndex] / 128);
        lastIndexSent = beatIndex;
        return result;
    }

    public void setBeat(int[] beat) {
        int index = beat[0];
        this.timeForNextBeat[index] = (short) beat[1];
        this.amplitudeForNextBeat[index] = (short) beat[2];
        this.drumNumber[index] = (short) beat[3];
    }

    public void setBeats(int[][] beats) {
        beatFrameLength = beats.length;
        for (int counter = 0; counter < beats.length; counter++) {
            setBeat(beats[counter]);
        }
    }
}
